src/app/clusters/power-source-configuration-server/power-source-configuration-server.cpp

```diff
@@ -61,9 +70,23 @@ CHIP_ERROR PowerSourceConfigurationAttrAccess::Read(const ConcreteReadAttributeP
     {
     case Sources::Id:
         err = aEncoder.EncodeList([](const auto & encoder) -> CHIP_ERROR {
+            uint16_t orderEpPair[kMaxPowerSources * 2];
+            uint8_t idx = 0, order = 0;
+            memset(orderEpPair, 0, sizeof(orderEpPair));
             for (auto endpoint : EnabledEndpointsWithServerCluster(PowerSource::Id))
             {
-                ReturnErrorOnFailure(encoder.Encode(endpoint));
+                if (idx >= sizeof(orderEpPair))
+                    break;
+                PowerSource::Attributes::Order::Get(endpoint, &order);
+                orderEpPair[idx]   = (uint16_t) order;
+                orderEpPair[++idx] = (uint16_t) endpoint;
+                idx++;
+            }
+
+            std::qsort(orderEpPair, idx / 2, sizeof(uint16_t) * 2, compareOrder);
+            for (uint8_t i = 0; i < idx; i += (uint8_t) 2)
```

Comment:
With an array of std::pair this could use a ranged for instead of casting in ways that might fail anyway on some compilers...